tangram/src/lib.rs

```diff
@@ -0,0 +1,559 @@
+
+
+
+	/// Total Token in system
+	/// pub Total get(fn total_items): TotalIndex;
+	#[pallet::storage]
+	#[pallet::getter(fn total_items)]
+	pub(super) type Total<T: Config> = StorageValue<_, TotalIndex, ValueQuery>; 
+
+
+
+	/// Burned Token in system
+	/// pub Burned get(fn burend_items): BurnedIndex;
+	#[pallet::storage]
+	#[pallet::getter(fn burned_items)]
+	pub(super) type Burned<T: Config> = StorageValue<_, BurnedIndex, ValueQuery>; 
+
+
+
+//-----------------------------news 
+
+
+	impl<Hash> TangramRealm <Hash> {
+		pub fn new(id: Hash, org: Hash, index: u64) ->  TangramRealm<Hash> where Hash: Clone, {
+			TangramRealm {
+				id,
+				org: org,
+				index: index,
+			}
+		}
+	}
+
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+	
+
+		#[pallet::weight(<T as Config>::WeightInfo::create_realm())]
+		pub fn create_realm(
+			origin: OriginFor<T>,
+			org: T::Hash
+		) -> DispatchResult {
+			// TODO: ensure origin == controller
+			// ensure_root(origin)?;
+
+			// TODO: ensure org exists : How???
+			// TODO: ensure org does not have a realm yet : ? Can org have 2 realms???
+			// ensure!(!<TangramRealm<T>>::contains_key(&account), Error::<T>::EntityExists);
+
+			let index = <NextRealmIndex<T>>::get();
+			let (hash, _blockNumber) = <T as Config>::Randomness::random(b"rndrealm");
+			let realm = TangramRealm::new(hash.clone(), org, index.clone() );
+			<Realm::<T>>::insert( hash.clone(), realm );
+			<RealmByIndex::<T>>::insert( index.clone(), hash );
+			// unsafe add
+			<NextRealmIndex<T>>::mutate(|i| *i += 1);
+			Self::deposit_event( Event::RealmCreated( index.clone() ) );
+			Ok(())
+		}
+
+
+		#[pallet::weight(<T as Config>::WeightInfo::create_class())]
+		pub fn create_class(
+			origin: OriginFor<T>,
+			realm: RealmIndex,
+			name: Vec<u8>,
+			max: u64,
+			strategy: u64
+		) -> DispatchResult {
+			// TODO: ensure origin == realm controller
+			// ensure_root(origin)?;
+	
+			// valid realm
+			let realm_index = <NextRealmIndex<T>>::get();
+			ensure!( realm < realm_index, Error::<T>::UnknownRealm );
+	
+			// max items
+			ensure!( max > 0, Error::<T>::MaxItemsTooSmall );
+			let index = Self::next_class_index(&realm);
+			<MaxItems<T>>::insert((&realm,&index),max.clone());
+	
+			// class
+			let (hash, _bn) = <T as Config>::Randomness::random(b"rndclass");
+			let new_class = TangramClass {
+				id: hash.clone(),
+				realm: realm.clone(),
+				index: index.clone(),
+				max: max.clone(),
+				strategy: 0 // (1,1,1)
+			};
+
+			<ItemClass::<T>>::insert( hash.clone(), new_class );
+			// ClassByIndex::<T>::insert( index.clone(), hash );
+	
+			<NextItemIndex<T>>::insert((&realm,&index),0);
+	
+			// ++ class index
+			let mut next_index = index.checked_add(1).ok_or(Error::<T>::Overflow)?;
+			<NextClassIndex<T>>::insert(&realm,next_index.clone());
+	
+			Self::deposit_event(Event::ClassCreated(realm, index, max));
+			Ok(())
+		}
+
+
```

Comment:
Please, add docs comments for every extrinsic. You can find examples in the other pallets.